packages/apputils/src/mainareawidget.ts

```diff
@@ -222,6 +226,7 @@ class MainAreaWidget<T extends Widget = Widget> extends Widget {
    * Give focus to the content.
    */
   private _focusContent(): void {
+    // TODO: perhaps we give the widget a chance to activate, and if that doesn't change the focus, *then* we focus the content node.
```

Comment:
still TODO?

Reply:
I think it's fine as is.

With the current order, we focus right away, but then post an asynchronous message to let the content widget do what it wants (which may include moving the focus somewhere else, of course). So the immediate result is that things are focused, but the content widget can override it.

With the other way, we'd really want to *send* the message (not post it) to give the content a chance to focus whatever it wants synchronously, and then if it doesn't focus something inside the content, force the focus on the content node.

I like the current order better: (a) it ensures that the focus actually is on the content node when the function exits, and (b) it gives the content a chance to override it asynchronously.


I'm basically asking if we should switch the order of these two cases.

With the current order, callig _focusContent could actually not focus the content, since the content's activate request could actually focus something else. However, maybe that is okay, in that we don't want to *force* having the content focused?

Reply:
I didn't want to force synchronous focusing down the widget chain.